### PR TITLE
Add kafka error handling bbe

### PR DESCRIPTION
**Pull request by dilanSachi**
## Purpose
$subject
https://github.com/ballerina-platform/ballerina-standard-library/issues/3804
Found an issue related to this,
https://github.com/ballerina-platform/ballerina-standard-library/issues/3884

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: examples/kafka-service-error-handling/kafka_service_error_handling.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Kafka service - Error handling
+
+When the `kafka:Service` encounters an internal error before invoking the `onConsumerRecord` method, such as a payload binding error or a constraint validation error, the `onError` method will be invoked with the relevant error. This error can be handled according to the user's need inside the `onError` method. If an error is returned from the `onConsumerRecord` method, it will be logged to the console, and the next polling cycle will continue.
```
Comment (MohamedSabthar) [inline examples/kafka-service-error-handling/kafka_service_error_handling.md:3]: How about something like the below

The `kafka:Service` has an `onError` method that is called when an internal error occurs before the `onConsumerRecord` method is called. These errors could include payload binding errors or constraint validation errors. The `onError` method allows the user to handle the error as needed. If an error is returned from the `onConsumerRecord` method, it is logged to the console and the next polling cycle continues.

Comment (shafreenAnfar) [reply examples/kafka-service-error-handling/kafka_service_error_handling.md:3]: Yeah, I like the suggestion.

Comment (dilanSachi) [reply examples/kafka-service-error-handling/kafka_service_error_handling.md]: Updated https://github.com/ballerina-platform/ballerina-distribution/pull/3903/commits/e7b60a94d353bcea1a2ecfb2417f3210e81eb474

File: examples/kafka-service-error-handling/kafka_service_error_handling.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Kafka service - Error handling
+
+The `kafka:Service` has an `onError` method that is called when an internal error occurs before the `onConsumerRecord` method is called. These errors could include payload binding errors or constraint validation errors. The `onError` method allows the user to handle the error as needed. If an error is returned from the `onConsumerRecord` method, it is logged to the console, and the next polling cycle continues.
```
Comment (shafreenAnfar) [inline examples/kafka-service-error-handling/kafka_service_error_handling.md:3]: The kafka:Service has an onError method which is invoked when `kafka:Listener` triggers errors  before the onConsumerRecord method is invoked. These errors could include payload binding errors or constraint validation errors. The onError method allows the user to handle these errors as needed. If there is no `onError` these are logged to the console with the stack-trace. In addition, if an error is returned from the onConsumerRecord method, it is logged to the console and the next polling cycle continues.

Comment (dilanSachi) [reply examples/kafka-service-error-handling/kafka_service_error_handling.md]: Updated

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**